### ISSUE TITLE
KeePassXC 2.7.7: keyfile input text is missing by default

### ISSUE THREAD

**Issue by samarlyka** [keepassxreboot/keepassxc#10551]
## Overview
In release dd21def (Version 2.7.6), opening a KeePassXC database would prompt both a password string input and a keyfile input. Strings in both fields are hidden (i.e., shown as "\*\*\*\*\*\*\*\*" instead of the actual string), which increases security.

However, since version 2.7.7, the keyfile input field is replaced with **"I have a key file"** button which pops up a file selector dialog. The file selection dialog clearly shows what folder is being opened and where the keyfile is located. In my perspective, this increases the risk of [shoulder surfing](https://en.wikipedia.org/wiki/Shoulder_surfing_(computer_security)).

If we allow the keyfile input text to be displayed (which hides actual string text just like password), **it would prevent [shoulder surfing](https://en.wikipedia.org/wiki/Shoulder_surfing_(computer_security))** so that someone in the same room who spies on our computer would not know where the keyfile is located, even if s/he attentively watches over our computer screen.

## Steps to Reproduce
1. Open a KeePassXC database encrypted using both password and keyfile from the command line (e.g., `/bin/keepassxc ~/keepassxc/Passwords.kdbx`)
2. KeePassXC only shows the password field, but not the keyfile field

![keepassxc2 7 7](https://github.com/keepassxreboot/keepassxc/assets/93555329/1b7068c9-c039-4767-a62b-d0e5e8a7f01f)

## Expected Behavior
N/A

## Actual Behavior
N/A

## Context
N/A

## Debug info
KeePassXC - Version 2.7.7
Revision: 68e2dd8

Qt 5.15.13
Debugging mode is disabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 6.8.2-arch2-1

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 3.3.0

## Computer environment
Operating System: Linux
Desktop Env: N/A (using i3-gaps without DE)
Windowing System: X11

**Comment by droidmonkey** [comment]
Not sure what the problem is? You need to pick a file to use it. That behavior didn't change between 2.7.6 and 2.7.7. Shoulder surfing is meaningless unless that surfer gains access to your computer with an unlocked database and/or knows your password. If you are worried about key file leakage, we strongly recommend using a hardware key instead.

**Comment by phoerious** [comment]
If you previously selected a keyfile, KeePassXC will remember it, unless you have that setting disabled.

**Comment by samarlyka** [comment]
> You need to pick a file to use it. That behavior didn't change between 2.7.6 and 2.7.7.

I have tried specifying the keyfile location using the following command:

    keepassxc --keyfile /path/to/keyfile PasswordDB.kdbx

My database has both keyfile and password set up.

In **2.7.6**, the above command prompted KeePassXC for a password while keeping the keyfile location intact.
But in **2.7.7**, KeePassXC simply assumes the database doesn't have any password set and directly attempts to unlock the database (which failed).
After the failed attempt, KeePassXC removes the history of keyfile location I specified in the command line.
Then I have to specify the password and repick the keyfile again to unlock the database.

> Not sure what the problem is?

I think that's the bug in **2.7.7**: using `--keyfile` flag does not prompt the user for password, even when the database is both password- and keyfile-encrypted.

**EDIT:** This issue (#10551) is about shoulder surfing. I'll create another issue which points out the **`--keyfile` flag does not prompt password** bug.

**EDIT 2:** The bug is now reported in #10552
